system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs

```diff
@@ -1033,8 +1043,8 @@ pub type Block = generic::Block<Header, UncheckedExtrinsic>;
 pub type SignedBlock = generic::SignedBlock<Block>;
 /// BlockId type as expected by this runtime.
 pub type BlockId = generic::BlockId<Block>;
-/// The SignedExtension to the basic transaction logic.
-pub type SignedExtra = (
+/// The TransactionExtension to the basic transaction logic.
+pub type TxExtension = (
```

Comment:
Why didn't we use `StorageWeightReclaim` here in the past?

We can use either one of:
* `StorageWeightReclaim` in crate `cumulus-pallet-weight-reclaim`. This is a replacement of `StorageWeightReclaim` in `cumulus-primitive-storage-weight-reclaim`, it uses the node to query the used proof size and reclaim the unused weight accordingly.
* `WeightReclaim` in crate `frame-system` it simply look at the unspent weight from transaction extensions and reduce the block weight.

I think we can add `WeightReclaim` in this PR. Later if wanted we can start using `StorageWeightReclaim`.

Reply:
I noticed `WeightReclaim` was introduced in `frame-system@40.1.0` (`stable2503`). So, I'll be skipping this for now, because we're a bit time-strechted to release this upgrade.